pkg/messaging/pubsub.go

```diff
@@ -23,14 +34,21 @@ type MessageHandler interface {
 	Cancel() error
 }
 
+type SubscriberConfig struct {
+	ID             string
+	Topic          string
+	Handler        MessageHandler
+	DeliveryPolicy DeliveryPolicy
+}
+
 // Subscriber specifies message subscription API.
 type Subscriber interface {
 	// Subscribe subscribes to the message stream and consumes messages.
-	Subscribe(ctx context.Context, id, topic string, handler MessageHandler) error
+	Subscribe(ctx context.Context, cfg SubscriberConfig) error
 
 	// Unsubscribe unsubscribes from the message stream and
 	// stops consuming messages.
-	Unsubscribe(ctx context.Context, id, topic string) error
+	Unsubscribe(ctx context.Context, cfg SubscriberConfig) error
```

Comment:
Let's keep ID and topic for now, since `cfg` can be a little confusing in the context of `Unsubscribe` (and other fields are not used and probably won't be used in the future, either).